src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -541,11 +623,41 @@ def list_recovery_points(cmd, client, resource_group_name, vault_name, item, sta
     if use_secondary_region:
         client = recovery_points_crr_cf(cmd.cli_ctx)
 
-    # Get recovery points
-    recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri, filter_string)
-    paged_recovery_points = _get_list_from_paged_response(recovery_points)
+    if recommended_for_archive:
+        client = recovery_points_recommended_cf(cmd.cli_ctx)
+        recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri)
+        paged_recovery_points = _get_list_from_paged_response(recovery_points)
 
-    return paged_recovery_points
+    else:
+        recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri,
+                                      filter_string)
+        paged_recovery_points = _get_list_from_paged_response(recovery_points)
+
+    fetch_tier(paged_recovery_points)
+
+    recovery_point_list = check_rp_move_readiness(paged_recovery_points, target_tier, is_ready_for_move)
+    recovery_point_list = filter_rp_based_on_tier(recovery_point_list, tier)
+    return recovery_point_list
+
+
+def move_recovery_points(cmd, resource_group_name, vault_name, item_name, rp_id, source_tier,
+                         destination_tier):
+
+    container_uri = _get_protection_container_uri_from_id(item_name.id)
+    item_uri = _get_protected_item_uri_from_id(item_name.id)
+
+    Dict = dict({'VaultStandard': 'HardenedRP', 'VaultArchive': 'ArchivedRP', 'Snapshot': 'InstantRP'})
```

Comment:
please don't do generic naming. give a name that is specific to this case and easily understandable.

Reply:
does 'tier_mapping' sound good (in place of Dict) ?

Reply:
use 'tier_type_map'

Reply:
done